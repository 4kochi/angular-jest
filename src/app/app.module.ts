import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { RouterModule } from '@angular/router';
import { AppComponent } from './app.component';
import { TodosModule } from './todos/todos.module';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, RouterModule.forRoot([]), TodosModule],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule {}
