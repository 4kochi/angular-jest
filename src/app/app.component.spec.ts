import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AppComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'todo-angular'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toEqual('todo-angular');
  });

  it('should render title in a h1 tag', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain('Welcome to todo-angular!');
    expect(compiled.querySelector('ul')).toMatchSnapshot();
    expect(compiled.querySelector('h2')).toMatchInlineSnapshot(`
<h2>
  Here are some links to help you start:
</h2>
`);
  });
});
