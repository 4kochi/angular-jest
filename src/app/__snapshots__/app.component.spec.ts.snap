// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render title in a h1 tag 1`] = `
<ul>
  <li>
    <h2>
      <a
        href="https://angular.io/tutorial"
        rel="noopener"
        target="_blank"
      >
        Tour of Heroes
      </a>
    </h2>
  </li>
  <li>
    <h2>
      <a
        href="https://angular.io/cli"
        rel="noopener"
        target="_blank"
      >
        CLI Documentation
      </a>
    </h2>
  </li>
  <li>
    <h2>
      <a
        href="https://blog.angular.io/"
        rel="noopener"
        target="_blank"
      >
        Angular blog
      </a>
    </h2>
  </li>
</ul>
`;
